### ISSUE TITLE
Current sensors should be float

### ISSUE THREAD

**Issue by mutzl** [syssi/homeassistant-goecharger-mqtt#58]
As already commented in the HA Forum, this integration treats Current values as integers, not as floats.
I see 2 problems here:

1. In this range >6 A and < 32 A, the first decimals do make a difference.
So I believe it should be `extract_item_from_array_to_float`,
<img width="330" alt="image" src="https://user-images.githubusercontent.com/2855185/202278126-e373a98c-ffb6-4209-b0a8-254ca84d8bf0.png">

2. Anyway, usage of int seems to be problematic to me when parsing json, as it just cuts the decimal part instead of rounding.
So something like this would make more sense, wouldn't it?
```
def extract_item_from_array_to_int(value, key) -> int:
    """Extract item from array to int."""
    return round(float(json.loads(value)[int(key)]))
```

Though I have never ever developed in Python before. So maybe this is complete nonsene.

**Comment by syssi** [comment]
Thanks for your detailed bug report! Could you add the `nrg` payload example here too? I will prepare a fix!

**Comment by mutzl** [comment]
![image](https://user-images.githubusercontent.com/2855185/202283475-c5485ed1-e4aa-4d2b-b7d9-15de694e8e3f.png)

**Comment by syssi** [comment]
I would be happy about a plain text copy of the response. :-) This is the response of my charger:

```
/go-eCharger/072246/nrg [232,232,233,1,0,0,0,0,0,0,0,0,0,0,0,0]
```

It looks like the resolution has changed / is different per model?

**Comment by syssi** [comment]
The plaintext copy of your response:

```
/go-eCharger/203597/nrg [232.5,231.26,234.67,2.48,5.966,6.004,5.996,1347.132,1358.567,1360.781,0,4066.48,97.11894,97.84528,95.1958,100]
```

**Comment by mutzl** [comment]
```
[
    233.43,
    231.26,
    234.67,
    2.48,
    5.928,
    6.004,
    5.966,
    1346.888,
    1354.837,
    1359.388,
    0,
    4061.113,
    97.33447,
    97.57664,
    97.0963,
    100
]
```

Firmware 054.7
Hardwareversion V4
Firmware-Version des Ladecontrollers 0.0.1-36

**Comment by syssi** [comment]
In my case it's firmware 054.6, hardware version 3, module revision 1 and charger controller version 12.

**Comment by mutzl** [comment]
> I've published a new release (`0.14.3`). Please give it a try! It hopefully solves your issue.

Works! 
<img width="264" alt="image" src="https://user-images.githubusercontent.com/2855185/202289048-905dfe69-0898-4be2-9f57-849df3c581c2.png">

Thank you!
